src/main/java/seedu/address/logic/commands/AddCommand.java

```diff
@@ -60,10 +76,96 @@ public CommandResult execute(Model model) throws CommandException {
+
+            List<Person> filteredByClassCodeList = studentList.stream()
+                    .filter(student -> (student.getClassCode().toString().equals(addedPerson.getClassCode().toString())
+                            || student.hasSameActivity(addedPerson))
+                            && !student.isSamePerson(addedPerson)
+                            && !student.getStatus().toString().equals(Status.POSITIVE))
+                    .collect(Collectors.toList());
+
+            for (int i = 0; i < filteredByClassCodeList.size(); i++) {
+                Person currentPerson = filteredByClassCodeList.get(i);
+                EditCommand.EditPersonDescriptor tempDescriptor = new EditCommand.EditPersonDescriptor();
+                tempDescriptor.setStatus(new Status(Status.CLOSE_CONTACT));
+                Person editedPersonStatus = createEditedPerson(currentPerson, tempDescriptor);
+                model.setPerson(currentPerson, editedPersonStatus);
+            }
+        } else {
+            List<Person> filteredByClassCodeAndActivityList = studentList.stream()
+                    .filter(student -> (student.getClassCode().toString()
+                            .equals(addedPerson.getClassCode().toString())
+                            || student.hasSameActivity(addedPerson))
+                            && !student.isSamePerson(addedPerson))
+                    .collect(Collectors.toList());
+
+            for (int i = 0; i < filteredByClassCodeAndActivityList.size(); i++) {
+                Person currentPerson = filteredByClassCodeAndActivityList.get(i);
+
+                List<Person> positiveRelatedToPerson = studentList.stream()
+                        .filter(student -> (student.getClassCode().toString()
+                                .equals(currentPerson.getClassCode().toString())
+                                || student.hasSameActivity(currentPerson))
+                                && !student.isSamePerson(addedPerson)
+                                && student.getStatus().toString().equals(Status.POSITIVE))
+                        .collect(Collectors.toList());
+
+                if (positiveRelatedToPerson.size() == 0) {
+                    EditCommand.EditPersonDescriptor tempDescriptor = new EditCommand.EditPersonDescriptor();
+                    tempDescriptor.setStatus(new Status(Status.NEGATIVE));
+                    Person editedPersonStatus = createEditedPerson(currentPerson, tempDescriptor);
+                    model.setPerson(currentPerson, editedPersonStatus);
+                } else {
+                    EditCommand.EditPersonDescriptor tempDescriptor = new EditCommand.EditPersonDescriptor();
+                    tempDescriptor.setStatus(new Status(Status.CLOSE_CONTACT));
+                    Person editedPersonStatus = createEditedPerson(addedPerson, tempDescriptor);
+                    model.setPerson(addedPerson, editedPersonStatus);
+                }
```

Comment:
code looks okay albeit a little long. Can leave it like that. but I think if we are to improve on the app further might need to abstract this out

Reply:
Agreed, we can disucss this on v1.4 and see what we can do about this 😄